packages/blockchain/test/pos.spec.ts

```diff
@@ -89,29 +88,6 @@ describe('Proof of Stake - inserting blocks into blockchain', () => {
         BigInt(1313601),
         'should have calculated the correct post-Merge total difficulty'
       )
-
-      const common = new Common({ chain: Chain.Mainnet, hardfork: Hardfork.London })
-      const powBlock = createBlockFromBlockData(
-        {
-          header: {
-            number: 16,
-            difficulty: BigInt(1),
```

Comment:
Ok.

Needed to think three times (or so 😋) about this.

So this is where this actually interflicts with our removed difficulty check.

So: a PoW block for Londond is created: works. Block is added to Blockchain: now (with removed check) works as well.

I wanted to suggest that we instead do a hardfork check on the Commons (block + blockchain).

This gets hairy quickly though (might still be the better thing to do), since we have this `_hardforkByHeadBlockNumber` setting in Blockchain and this generally reaches deep into the blockchain update code.

I will capitulate here for now and re-add the difficulty check for now, together with a comment that this is not really a consensus check. That should enable to re-activate the test case.

Reply:
Yes, works (locally), and I guess for this PR it is ok (maybe: better), since this PR is about something else at the end.

But to admit, this difficulty check is just picking out a very selected case and this problem that it should not be possible to add a non-HF matching block (with eventually different formatting or properties) to a blockchain is a lot broader and should likely be solved on a more generic level.